pkgs/applications/backup/pika-backup/default.nix

```diff
@@ -0,0 +1,77 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, rustPlatform
+, substituteAll
+, desktop-file-utils
+, meson
+, ninja
+, pkg-config
+, python3
+, wrapGAppsHook
+, borgbackup
+, dbus
+, gdk-pixbuf
+, glib
+, gtk3
+, libhandy
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pika-backup";
+  version = "0.2.1";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "pika-backup";
+    rev = "v${version}";
+    sha256 = "0fm6vwpw0pa98v2yn8p3818rrlv9lk3pmgnal1b2kh52im5ll7m8";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    sha256 = "1f5s6a0wjrs2spsicirhbvb5xlz9iflwsaqchij9k02hfcsr308y";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./borg-path.patch;
+      borg = "${borgbackup}/bin/borg";
+    })
+  ];
```

Comment:
This can probably be replaced with ``substituteInPlace src/borg/utils.rs b/src/borg/utils.rs --replace 'Command::new("borg");' "Command::new(${borgbackup}/bin/borg);"``

Reply:
It can. But that makes it less obvious when things break. That's why I prefer patches.

Reply:
Substitute in place logs a warning when it can't find the pattern which nixpkgs hammering catches. Patches for simple things should be avoided because they clutter the repository and blow up the file count and size.

Reply:
On the other hand, they are much easier to manage – you can simply apply them to the source code using `git apply` and then re-generate them using `git diff`. Manual substitute is more laborious and thus more prone to mistakes.

Reply:
I think there are arguments in favor of both approaches. So, I think we should favor what the package maintainer prefers. They have to maintain the patches/substitutions after all.